src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -245,21 +247,30 @@ private void initHttpAndDiskCache(
     RemoteCache remoteCache = new RemoteCache(cacheClient, remoteOptions, digestUtil);
     actionContextProvider =
         RemoteActionContextProvider.createForRemoteCaching(
-            executorService, env, remoteCache, /* retryScheduler= */ null, digestUtil);
+            executorService,
+            env,
+            remoteCache,
+            /* retryScheduler= */ null,
+            digestUtil,
+            tempPathGenerator);
   }
 
   @Override
   public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
     Preconditions.checkState(actionContextProvider == null, "actionContextProvider must be null");
     Preconditions.checkState(actionInputFetcher == null, "actionInputFetcher must be null");
     Preconditions.checkState(remoteOptions == null, "remoteOptions must be null");
+    Preconditions.checkState(tempPathGenerator == null, "tempPathGenerator must be null");
 
     RemoteOptions remoteOptions = env.getOptions().getOptions(RemoteOptions.class);
     if (remoteOptions == null) {
       // Quit if no supported command is being used. See getCommandOptions for details.
       return;
     }
 
+    this.tempPathGenerator =
```

Comment:
We need to clear stale tmp files just in case something went wrong. Please check the code here https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/remote/AbstractActionInputPrefetcher.java;l=81;drc=cfea5d36056c71da49984d511262ed18a596862b.

Reply:
Done. I also had to move the `TmpPathGenerator` instantiation to `RemoteModule#executorInit`. `RemoteModule#beforeCommand` is too early to do that; we need the execroot to be known, which in turn needs the workspace name to have been parsed from the `WORKSPACE` file.